pkg/clients/gitlab/git.go

```diff
@@ -194,3 +194,23 @@ func (gc *GitlabClient) ValidateNoteInMergeRequestComment(projectID, expectedNot
 		return false
 	}, timeout, interval).Should(BeTrue(), fmt.Sprintf("timed out waiting to validate merge request note ('%s') be reported in mergerequest %d's notes", expectedNote, mergeRequestID))
 }
+
+// IsWebhookExists checks if a webhook exists in the given project and if its URL contains the clusterAppDomain.
+// It returns true if such a webhook exists, otherwise false.
+func (gc *GitlabClient) IsWebhookExists(projectID, clusterAppDomain string) (bool, error) {
+
+	// List all project hooks
+	webhooks, _, err := gc.client.Projects.ListProjectHooks(projectID, nil)
+	if err != nil {
+		return false, fmt.Errorf("failed to list project hooks: %v", err)
+	}
+
+	// Check if any webhook URL contains the clusterAppDomain
+	for _, webhook := range webhooks {
+		if strings.Contains(webhook.URL, clusterAppDomain) {
+			return true, nil
+		}
+	}
```

Comment:
IIUC, we're already doing this within the [DeleteWebhooks() function](https://github.com/konflux-ci/e2e-tests/blob/main/pkg/clients/gitlab/git.go#L133-L147).